src/mavros_drone.py

```diff
@@ -0,0 +1,235 @@
+        if not self.location:
+            return {"success": False, "message": "Failed to upload " +
+                    "waypoints. Drone location is unknown."}
+        self.waypoints = waypoints
+
+        # Converts all the NavSatFix messages to Waypoint so that
+        # they're MAVROS compatible
+        converted_waypoint_objects = []
+        for navsatfix in waypoints:
+            converted_waypoint_objects.append(
+                self.convert_navsatfix_mavroswaypoint(navsatfix))
+
+        # Two takeoff commands prepended to the waypoint list for safety
+        converted_waypoint_objects = 2 * [
+            {'frame': MavrosDrone.FRAME_REFERENCE.RELATIVE_ALT.value,
+            'command': MavrosDrone.MAV_CMD.TAKEOFF.value, 'is_current': False,
+            'autocontinue': True, 'param1': 0, 'param2': 0, 'param3': 0,
+            'x_lat': self.location['latitude'],
+            'y_long': self.location['longitude'],
+            'z_alt': 10}
+            ] + converted_waypoint_objects
+        print(converted_waypoint_objects)
+
+        try:
+            print("Attempting to upload mission...")
+            service = roslibpy.Service(self.ROS_master_connection,
+                                       '/mavros/mission/push',
+                                       'mavros_msgs/WaypointPush')
+            request = roslibpy.ServiceRequest(
+                {'waypoints': converted_waypoint_objects})
+
+            print('Calling /mavros/mission/push service...')
+            result = service.call(request)
+            print('Service response: {}'.format(result))
+        except:
+            result = {"success": False,
+                      "message": "Failed to upload waypoints"}
+
+        return result
+
+    def convert_navsatfix_mavroswaypoint(self, navsatfix):
+        '''
+        Takes in a NavSatFix message and returns a
+        mavros_msgs/Waypoint message as a dictionary.
+        '''
+        waypoint = {'frame': MavrosDrone.FRAME_REFERENCE.RELATIVE_ALT.value,
+                    'command': MavrosDrone.MAV_CMD.NAVIGATE_TO_WAYPOINT.value,
+                    'is_current': False, 'autocontinue': True, 'param1': 0,
+                    'param2': 0, 'param3': 0, 'x_lat': navsatfix['latitude'],
+                    'y_long': navsatfix['longitude'],
+                    'z_alt': navsatfix['altitude']}
+
+        return waypoint
+
+
+    def set_speed(self, speed):
+        try:
+            print("Attempting to set speed...")
+            service = roslibpy.Service(self.ROS_master_connection,
+                                       '/mavros/cmd/command',
+                                       'mavros_msgs/CommandLong')
+            request = roslibpy.ServiceRequest(
+                {"command": MavrosDrone.MAV_CMD.SET_SPEED.value,
+                 "param1": 0, "param2": speed, "param3": -1, "param4": 0})
+            print('Calling mission_waypoint_setSpeed service...')
+            result = service.call(request)
```

Comment:
Does the result of this service request have the form {success, message}?
Ditto at all the other services...